src/ControlSystem/Protocols/ControlError.hpp

```diff
@@ -0,0 +1,42 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+#include <type_traits>
+
+#include "DataStructures/DataVector.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace control_system::protocols {
+/// \brief Definition of a control error
+///
+/// A control error is used within a control system to compute how far off the
+/// the value you are controlling is from its expected value.
+///
+/// A conforming type must specify:
+///
+/// - a call operator that returns a DataVector with a signature the same as in
+///   the example shown here:
+///   \snippet Helpers/ControlSystem/Examples.hpp ControlError
+struct ControlError {
+  template <typename ConformingType>
+  struct test {
+    struct DummyMetavariables;
+    struct DummyTupleTags;
+
+    static_assert(
+        std::is_same_v<
+            DataVector,
+            decltype(ConformingType{}(
+                std::declval<
+                    const Parallel::GlobalCache<DummyMetavariables>&>(),
+                std::declval<const double>(),
+                std::declval<const std::string&>(),
+                std::declval<const tuples::TaggedTuple<DummyTupleTags>&>()))>);
```

Comment:
Question to make sure I understand what will happen for size control:  Suppose size control needs a horizon, an excision boundary, and also lapse, shift, and 3-metric on the excision boundary.  So will there be three DummyTupleTags?  (two Strahlkorpers [AhA and ExcisionBdryA] and a Variables containing lapse+shift+3-metric)?    Or, because the excision boundary is one "measurement", does the excision boundary measurement need to return a struct that contains a Strahlkorper plus a Variables?  (so then there will be only two DummyTupleTags?)

Reply:
That will depend on how the size-control measurement is set up. If you could somehow retrieve the lapse, shift and 3-metric from just the excision boundary (EB) strahlkorper (unlikely) then you'd only need two DummyTupleTags (because you could compute the lapse+shift+3-metric inside the control error).

However, the more likely scenario is that finding the apparent horizon will be one [submeasurement](https://github.com/sxs-collaboration/spectre/blob/0e4e2956da06b1dbf3f7213e0ffd0c41feaf9b5f/src/ControlSystem/Protocols/Submeasurement.hpp), finding the EB will be another submeasurement, and getting the lapse+shift+3-metric in a Variables on the EB (an interpolation) is a third submeasurement. All within the size control measurement. So then you will need three DummyTupleTags like you said: two strahlkorpers and a Varaibles.

Reply:
I guess the reason I am confused is I don't understand what a submeasurement is.   Is a submeasurement really something associated with an interpolation_target_tag?  So if we have 3 submeasurements, we need 3 interpolation_target_tags? [note that the EB strahlkorper and lapse+shift+3metric on the EB strahlkorper all live on the same interpolation_target_tag].

Reply:
Recap of explanation with @markscheel and @nilsdeppe.
For size control, there will be two submeasurements. One associated with interpolating to an AH surface and doing a horizon find, the other associated with interpolating to the EB (and whatever needs to happen after the interpolation). Then `DummyTupleTags` are the info the control system picks from these submeasurements. So there will likely be 3: one for AH strahlkorper, one for EB strahlkorper, and one for a Variables on the EB.